packages/grafana-ui/src/components/Table/DefaultCell.tsx

```diff
@@ -101,28 +100,17 @@ function getCellStyle(
   disableOverflowOnHover = false,
   isStringValue = false
 ) {
-  // How much to darken elements depends upon if we're in dark mode
-  const darkeningFactor = tableStyles.theme.isDark ? 1 : -0.7;
-
   // Setup color variables
   let textColor: string | undefined = undefined;
   let bgColor: string | undefined = undefined;
 
-  if (cellOptions.type === TableCellDisplayMode.ColorText) {
-    textColor = displayValue.color;
-  } else if (cellOptions.type === TableCellDisplayMode.ColorBackground) {
-    const mode = cellOptions.mode ?? TableCellBackgroundDisplayMode.Gradient;
-
-    if (mode === TableCellBackgroundDisplayMode.Basic) {
-      textColor = getTextColorForAlphaBackground(displayValue.color!, tableStyles.theme.isDark);
-      bgColor = tinycolor(displayValue.color).toRgbString();
-    } else if (mode === TableCellBackgroundDisplayMode.Gradient) {
-      const bgColor2 = tinycolor(displayValue.color)
-        .darken(10 * darkeningFactor)
-        .spin(5);
-      textColor = getTextColorForAlphaBackground(displayValue.color!, tableStyles.theme.isDark);
-      bgColor = `linear-gradient(120deg, ${bgColor2.toRgbString()}, ${displayValue.color})`;
-    }
+  // Get colors
+  const colors = getCellColors(tableStyles, cellOptions, displayValue);
+  textColor = colors.textColor;
+  
+  // Skip application of background color if we're applying to the whole row
+  if (cellOptions.type === TableCellDisplayMode.ColorBackground && !cellOptions.applyToRow) {
+    bgColor = colors.bgColor;
```

Comment:
this is a bit of shame as it means we cannot override the color for specific fields. I guess we need to disable this to make the gradient not look strange (broken up)?

Reply:
Yep, this is to have the gradient stretch across the row as opposed to the row and the cell as well. Separate field overrides might still work. I can take a look there 😃